pkg/kn/commands/completion_helper.go

```diff
@@ -0,0 +1,129 @@
+// Copyright © 2021 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+type completionConfig struct {
+	params     *KnParams
+	command    *cobra.Command
+	args       []string
+	toComplete string
+	target     string
```

Comment:
I wonder whether `target` needs to be in an extra field. It's specific only to `service` commands and could easily be extracted from the `command` when it is already clear that we are dealing with services.

Reply:
yes, I think we can remove the field